observable.go

```diff
@@ -1111,6 +1112,70 @@ func (o *observable) Repeat(count int64, frequency Duration) Observable {
 	return newColdObservableFromFunction(f)
 }
 
+// Sample returns an Observable that emits the most recent items emitted by the source
+// ObservableSource whenever the input Observable emits an item.
+func (o *observable) Sample(obs Observable) Observable {
+	f := func(out chan interface{}) {
+		mainChan := make(chan interface{})
+		obsChan := make(chan interface{})
+		var lastEmittedItem interface{}
+		isItemWaitingToBeEmitted := false
+		ctx, cancel := context.WithCancel(context.Background())
+
+		go func() {
+			it := o.iterable.Iterator(ctx)
+			for {
+				if item, err := it.Next(ctx); err == nil {
+					mainChan <- item
+				} else {
+					break
+				}
+			}
+			close(mainChan)
+		}()
+
+		go func() {
+			it := obs.Iterator(ctx)
+			for {
+				if item, err := it.Next(ctx); err == nil {
+					obsChan <- item
+				} else {
+					break
+				}
+			}
+			close(obsChan)
+		}()
+
+	mainLoop:
```

Comment:
One very last comment and we can merge it. Can we get rid of this label and replace the two `break mainLoop` by `return`. I prefer avoiding label unless it is absolutely mandatory.